app/browser/reducers/siteSettingsReducer.js

```diff
@@ -79,6 +81,33 @@ const siteSettingsReducer = (state, action, immutableAction) => {
         }
         break
       }
+    case appConstants.APP_TAB_UPDATED:
+      {
+        let url = action.get('tabValue').get('url')
+        let parsedUrl = urlParse(url)
+        let currentSiteSetting = state.get('siteSettings')
+        let siteSafeBrowsing = true
+        let safeBrowsingAll = state.get('safeBrowsingAll')
+        let safeBrowsingAllSetting = true
+
+        if (safeBrowsingAll !== undefined) {
+          safeBrowsingAllSetting = safeBrowsingAll.get('enabled')
+        }
+
+        if (currentSiteSetting !== undefined &&
+          (parsedUrl.protocol === 'https:' || parsedUrl.protocol === 'http:')) {
+          let ruleKey = `https?://${parsedUrl.host}`
+          let currentSettings = currentSiteSetting.get(ruleKey)
+          if (currentSettings !== undefined) {
+            siteSafeBrowsing = currentSettings.get('safeBrowsing')
+          }
+        }
+        siteSafeBrowsing = siteSafeBrowsing === undefined ? true : siteSafeBrowsing
+
+        setUserPref('safebrowsing.enabled',
```

Comment:
You can't turn a global setting on/off per tab. I don't think it makes sense or is even possible in the current implementation to have site settings for safe browsing. With this implementation it could turn off for a tab loading in the background because the foreground tab has it disabled